projects/common/src/time/time.ts

```diff
@@ -32,11 +32,27 @@ export class Time {
     return this._date;
   }
 
+  public static parse(label: string, isUTC: boolean = false): Time {
+    const time: string[] = label.split(' ');
+    const meridian: string = time.pop()!;
+    const [hours, minutes, seconds, milliseconds] = time[0].split(':').map(value => +value);
```

Comment:
do we always get 4 components here? if not, there could be undefined values here.

Reply:
Today `label` only return "hh:mm merdian", but kept seconds and milliseconds here so that we don't need to change it in the future if we decide to show seconds and milliseconds too in the label or something like that. This method can handle that. We anyways assign 0 to minutes, seconds and milliseconds if they are undefined

Reply:
<img width="410" alt="image" src="https://user-images.githubusercontent.com/60038750/212288825-63dc31ce-b236-4966-9f30-7f75afae1413.png">

Reply:
I'm really questioning this use case - where are we getting a non ISO formatted date that we'd want to parse? This sounds like we're fixing that should be fixed downstream.